syntax = "proto3";

package metrics;


option go_package = "./proto/metrics/pb";

service MetricsService{
  rpc RequestTemp(TemperatureRequest) returns (TemperatureResponse) {}
  rpc RequestPol(PollutionRequest) returns (PollutionResponse) {}
}

message TemperatureResponse {
  int64 city = 1;
  string district = 2;
  int64 temperature = 3;
  int64 humidity = 4;
}

message PollutionResponse {
  int64 city = 1;
  string district = 2;
  int64 co2 = 3;
  int64 pm25 = 4;
}

message TemperatureRequest {
  int64 city = 1;
  string type = 2;
}

message PollutionRequest {
  int64 city = 1;
  string type = 2;
}